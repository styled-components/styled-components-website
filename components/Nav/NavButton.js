import styled from 'styled-components';

import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import { resetInput } from '../../utils/form';

const NavButton = styled.button`
  ${resetInput};
  background: ${p => (p.active ? 'rgba(0, 0, 0, 0.07)' : 'none')};
  cursor: pointer;
  flex: 0 0 auto;
  height: ${rem(navbarHeight)};
  padding: 0 ${rem(10)};
  text-align: center;
  vertical-align: middle;

  & + & {
    padding-left: 0;
  }
`;

export default NavButton;
