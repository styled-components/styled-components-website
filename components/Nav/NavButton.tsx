import styled from 'styled-components';
import { resetInput } from '../../utils/form';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';

const NavButton = styled.button<{ $active?: boolean }>`
  ${resetInput};
  background: ${p => (p.$active ? 'rgba(0, 0, 0, 0.07)' : 'none')};
  cursor: pointer;
  flex: 0 0 auto;
  height: ${rem(navbarHeight)};
  padding: 0 ${rem(10)};
  text-align: center;
  vertical-align: middle;

  & + & {
    padding-left: 0;
  }
`;

export default NavButton;
