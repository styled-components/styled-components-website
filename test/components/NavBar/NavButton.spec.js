import React from 'react';
import renderer from 'react-test-renderer';
import NavButton from '../../../components/Nav/NavButton';

test('NavButton renders correctly', () => {
  const tree = renderer.create(<NavButton />).toJSON();

  expect(tree).toMatchSnapshot();
});
