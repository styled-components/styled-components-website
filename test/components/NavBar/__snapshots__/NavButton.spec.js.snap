// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton renders correctly 1`] = `
.c1 {
  background: none;
  outline: none;
  border: none;
  background: none;
  cursor: pointer;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2.7777777777777777rem;
  padding: 0 0.5555555555555556rem;
  text-align: center;
  vertical-align: middle;
}

.c0 + .c0 {
  padding-left: 0;
}

<button
  className="c0 c1"
/>
`;
